core/types/blob_tx.go

```diff
@@ -47,15 +48,50 @@ func (stx BlobTx) GetBlobGas() uint64 {
 }
 
 func (stx BlobTx) AsMessage(s Signer, baseFee *big.Int, rules *chain.Rules) (Message, error) {
-	msg, err := stx.DynamicFeeTransaction.AsMessage(s, baseFee, rules)
-	if err != nil {
-		return Message{}, err
+	msg := Message{
+		nonce:      stx.Nonce,
+		gasLimit:   stx.Gas,
+		gasPrice:   *stx.FeeCap,
+		tip:        *stx.Tip,
+		feeCap:     *stx.FeeCap,
+		to:         stx.To,
+		amount:     *stx.Value,
+		data:       stx.Data,
+		accessList: stx.AccessList,
+		checkNonce: true,
+	}
+	if !rules.IsCancun {
```

Comment:
I suggest to keep the old `stx.DynamicFeeTransaction.AsMessage` but add the `!rules.IsCancun` check.

Reply:
Or that wouldn't work because of overridden `Sender`?

Reply:
No this will call signer.Sender as a DynamicFeeTransaction and as a result sender will be incorrect for some reason.